src/org/labkey/test/components/html/ToggleButton.java

```diff
@@ -0,0 +1,90 @@
+package org.labkey.test.components.html;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class ToggleButton extends WebDriverComponent<ToggleButton.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public ToggleButton(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return _driver;
+    }
+
+    public ToggleButton set(boolean enabled)
+    {
+        String desiredState = enabled ? "enabled" : "disabled";
+        if (get() != enabled)
+            getComponentElement().click();
+        getWrapper().waitFor(()-> get() == enabled,
```

Comment:
Pay attention to IntelliJ warnings. `waitFor` is a static method:
```suggestion
        WebDriverWrapper.waitFor(()-> get() == enabled,
```